src/dbnode/storage/namespace.go

```diff
@@ -1196,21 +1196,33 @@ func (n *dbNamespace) ColdFlush(flushPersist persist.FlushPreparer) error {
 		return err
 	}
 
+	// NB(bodu): Deferred shard cold flushes so that we can ensure that cold flush index data is
```

Comment:
Not part of this PR, but noticed that `n.metrics.flushColdData.ReportSuccessOrError(res, n.nowFn().Sub(callStart))` won't get called if we hit one of the errors above; is this correct behaviour?

Reply:
Hm, it should since "res = multiError.FinalError()"?

Reply:
Yeah I mean in the error paths above this